src/core/stac/qgsstacdataitems.cpp

```diff
@@ -202,6 +202,7 @@ void QgsStacItemItem::itemRequestFinished( int requestId, QString error )
   }
   else
   {
+    delete object;
```

Comment:
if setStacItem() would accept a unique_ptr there could be a way of avoiding manual deletion, but I realize this might lead to more changes.

Reply:
I've changed signatures to return/accept unique_ptr when ownership is transferred